src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/BuildOptions.cs

```diff
@@ -1,39 +1,49 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
 
 namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
 {
     internal class BuildOptions
     {
-        public IEnumerable<CommandLineSourceFile> SourceFiles { get; }
-        public IEnumerable<CommandLineSourceFile> AdditionalFiles { get; }
-        public IEnumerable<CommandLineReference> MetadataReferences { get; }
-        public IEnumerable<CommandLineAnalyzerReference> AnalyzerReferences { get; }
-
-        public BuildOptions(IEnumerable<CommandLineSourceFile> sourceFiles, IEnumerable<CommandLineSourceFile> additionalFiles, IEnumerable<CommandLineReference> metadataReferences, IEnumerable<CommandLineAnalyzerReference> analyzerReferences)
+        public BuildOptions(ImmutableArray<CommandLineSourceFile> sourceFiles, ImmutableArray<CommandLineSourceFile> additionalFiles, ImmutableArray<CommandLineReference> metadataReferences, ImmutableArray<CommandLineAnalyzerReference> analyzerReferences)
         {
-            Requires.NotNull(sourceFiles, nameof(sourceFiles));
```

Comment:
Removing this could cause an NRE [here](https://github.com/dotnet/project-system/blob/2e9680b08d1f05ea1f0a994ad42ae4757a1d65af/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs#L54).  While the C#/VB collections won't ever be null, that's merely by convention in the Roslyn code base and once a proper command line parser for F# is added, there's no guarantee there.